core/src/main/java/org/apache/iceberg/data/avro/GenericReaders.java

```diff
@@ -105,7 +109,10 @@ private static class GenericRecordReader extends ValueReaders.StructReader<Recor
     private final StructType structType;
 
     private GenericRecordReader(
-        List<ValueReader<?>> readers, StructType struct, Map<Integer, ?> idToConstant) {
+        List<ValueReader<?>> readers,
+        StructType struct,
+        Schema record,
+        Map<Integer, ?> idToConstant) {
```

Comment:
It looks like this change isn't needed, nor is the update to add `Schema` to `createStructReader`.

Reply:
Reverted.